sdk/identity/identity/src/client/identityClient.ts

```diff
@@ -41,7 +41,16 @@ export class IdentityClient extends ServiceClient {
 
   constructor(options?: TokenCredentialOptions) {
     if (isNode) {
-      options = options || getAuthorityHostEnvironment();
+      let authorityHostEnv = getAuthorityHostEnvironment();
```

Comment:
I personally would flip this, so we don't have to always check the env var - assuming that read from var is faster than reading from env

if(!options.authorityHost){
  // get from env
 // if not in env, use default.
}

Reply:
That wouldn't give you the same precedence. That would allow the options bag to override the environment (which is how it currently works)